aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -59,6 +59,40 @@ TORCH_META_FUNC(mm)(const Tensor & self, const Tensor & mat2) {
   set_output(0, {self.sizes()[0], mat2.sizes()[1]}, {}, self.options(), names);
 }
 
+TORCH_META_FUNC(linalg_vector_norm)(const Tensor& self, const Scalar& scalar_ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  at::native::checkFloatingOrComplex(self, "linalg.vector_norm");
+
+  auto dim = opt_dim.value_or(IntArrayRef{});
+  // Casting a large integer to a double will just introduce an error for
+  // values larger than 10^53 (same for negative numbers), so that's fine.
+  auto ord = scalar_ord.toDouble();
+
+  // If norm < 0 || norm == infty, we cannot reduce over an empty dimension (see issue 52783)
+  if (opt_dim.has_value() && self.numel() == 0 && (ord < 0. || ord == INFINITY)) {
+    for (auto dim_num : dim) {
+      TORCH_CHECK(self.size(dim_num) != 0,
+        "linalg.vector_norm cannot compute the ", scalar_ord, " norm on an empty ",
+        "dimension because the operation does not have an identity");
+    }
+  }
+
+  if (opt_dtype.has_value()) {
+    auto dtype = opt_dtype.value();
+    TORCH_CHECK(isFloatingType(dtype) || isComplexType(dtype), "linalg.vector_norm(): dtype should"
+        " be floating point or complex, but got ", dtype);
+    TORCH_CHECK(self.is_complex() == isComplexType(dtype),
+        "linalg.vector_norm(): dtype should be ", self.is_complex() ? "complex" : "real",
+        " for ", self.is_complex() ? "complex" : "real", " inputs.");
```

Comment:
A question we didn't discuss but I think it's fair is: should `self` be broadcastable without narrowing to `dtype`? I'd say it should, because that's the intended behaviour, but what do you think?

Reply:
Convertible you mean? Yeah that seems reasonable (although conversion would already warn when you convert complex to real)